gorapass/views.py

```diff
@@ -23,9 +35,49 @@ def hike(request, hike_id):
 
 def hikes(request):
     hike_models = Hikes.objects.all()
+
+    # Filter out hikes if there is filtration criteria in the request
+    if request.body:
+        filters = json.loads(request.body)['filters']
+        if not valid_hike_filters(filters):
+            return HttpResponseBadRequest('Invalid filtration criteria')
+
+        hike_models = filter_hikes(hike_models, filters)
+
     hike_dicts = [ model_to_dict(hike) for hike in hike_models ]
     return JsonResponse(hike_dicts, safe=False)
 
+def valid_hike_filters(filters):
+    for filter in filters:
+        if set(filter.keys()) != FILTER_KEYS:
+            return False
+        if not hasattr(Hikes, filter[ATTRIBUTE_NAME]):
+            return False
+        
+    return True
+
+def filter_hikes(hike_models, filters):
+    for filter in filters:
+        attribute = filter[ATTRIBUTE_NAME]
+        value = filter[ATTRIBUTE_VALUE]
+        match filter[FILTER_TYPE]:
```

Comment:
I haven't seen the `match` statement used before. This is cool. So basically, you are whittling down the list of hike_models with every loop to filter it down.

Reply:
Exactly. I think this is better than checking each hike for *all* filters.